### PR TITLE
GEM-134 | Add generate group image

### PR DESCRIPTION
**Pull request by lawos98**

### CODE REVIEW

File: src/main/kotlin/pl/edu/agh/gem/external/controller/InternalGroupController.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import pl.edu.agh.gem.external.dto.GroupAttachmentResponse
+import pl.edu.agh.gem.internal.service.GroupService
+import pl.edu.agh.gem.media.InternalApiMediaType.APPLICATION_JSON_INTERNAL_VER_1
+
+@RestController
+@RequestMapping("/internal/groups")
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/controller/InternalGroupController.kt]: please use INTERNAL constant from lib

File: src/main/kotlin/pl/edu/agh/gem/external/controller/InternalGroupController.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+class InternalGroupController(
+    val groupService: GroupService,
+) {
+    @PostMapping("/{groupId}/users/{userId}/generate", produces = [APPLICATION_JSON_INTERNAL_VER_1])
+    @ResponseStatus(CREATED)
+    fun saveAttachment(
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/controller/InternalGroupController.kt]: would rename to generateAttachment. (but save is ok)

File: src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt [hunk 3]
```diff
@@ -30,4 +30,10 @@ class ServiceTestClient(applicationContext: WebApplicationContext) {
             .headers { it.withValidatedUser(user) }
             .exchange()
     }
+
+    fun generateGroupAttachment(userId: String, groupId: String): ResponseSpec {
+        return webClient.post()
+            .uri(URI("/internal/groups/$groupId/users/$userId/generate"))
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt]: please use INTERNAL constant

File: src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt [hunk 4]
```diff
@@ -30,4 +30,10 @@ class ServiceTestClient(applicationContext: WebApplicationContext) {
...
+            .exchange()
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt:41]: shouldn't we add header: .withAppAcceptType() ?

**Review by pwgacek — CHANGES_REQUESTED** [review]

File: src/main/kotlin/pl/edu/agh/gem/external/controller/InternalGroupController.kt (hunk 2 above)
Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/external/controller/InternalGroupController.kt]: sure changed

File: src/main/kotlin/pl/edu/agh/gem/external/controller/InternalGroupController.kt (hunk 1 above)
Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/external/controller/InternalGroupController.kt]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt (hunk 3 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt (hunk 4 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt:41]: sure we can add that but that header with only check on client side

**Review by pwgacek — APPROVED** [review]

**Pull request merged**